arch/x86/core/ia32/gdbstub.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_internal.h>
+#include <ia32/exception.h>
+#include <inttypes.h>
+#include <debug/gdbstub.h>
+
```

Comment:
a lot of this code will also work on x86_64, consider moving this file up to arch/x86/core although I suppose we can do that in a later PR once x86_64 is supported